### PR TITLE
Update export.go (fix errcheck issue)

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28921

```release-note
NONE
```

### CODE REVIEW

File: server/.golangci.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ issues:
         channels/app/config_test.go|\
         channels/app/desktop_login.go|\
         channels/app/email/email_test.go|\
-        channels/app/export.go|\
+        channels/app/email/helper_test.go|\
```
Comment (hanzei) [inline server/.golangci.yml:100]: That seems wrong

File: server/channels/app/export.go [hunk 2]
```diff
@@ -869,7 +873,8 @@ func (a *App) exportCustomEmoji(c request.CTX, job *model.Job, writer io.Writer,
 		pathToDir := filepath.Join(outPath, exportDir)
 		if exportFiles {
 			if _, err := os.Stat(pathToDir); os.IsNotExist(err) {
-				os.Mkdir(pathToDir, os.ModePerm)
+				if err := os.Mkdir(pathToDir, os.ModePerm); err != nil {
+					c.Logger().Error("Error creating directory", mlog.String("path", pathToDir), mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/export.go:877]: I think we should return the error here. If we can't create the export dir, the rest of the method will just fail. Let's be load and clear.

File: server/channels/app/export.go [hunk 3]
```diff
@@ -902,7 +907,11 @@ func (a *App) copyEmojiImages(emojiId string, emojiImagePath string, pathToDir s
 	if fromPath == nil || err != nil {
 		return errors.New("Error reading " + emojiImagePath + "file")
 	}
-	defer fromPath.Close()
+	defer func() {
+		if err = fromPath.Close(); err != nil {
+			a.Log().Error("Error closing file", mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/export.go:912]: Can we add a `request.CTX` to `copyEmojiImages` and use it's `Logger()` instead of the global one? Thanks!

File: server/channels/app/export.go [hunk 4]
```diff
@@ -1155,15 +1171,19 @@ func (a *App) exportFile(outPath, filePath string, zipWr *zip.Writer) *model.App
...
-		wr, err = os.Create(filePath)
+		wr, err := os.Create(filePath)
 		if err != nil {
 			return model.NewAppError("exportFileAttachment", "app.export.export_attachment.create_file.error",
 				nil, "err="+err.Error(), http.StatusInternalServerError)
 		}
-		defer wr.(*os.File).Close()
+		defer func() {
+			if err = wr.(*os.File).Close(); err != nil {
+				a.Log().Error("Error closing file", mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/export.go:1181]: same here

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR :tada:

File: server/channels/app/export.go [hunk 5]
```diff
@@ -1130,14 +1142,18 @@ func (a *App) exportAllDirectPosts(ctx request.CTX, job *model.Job, writer io.Wr
 	return attachments, nil
 }
 
-func (a *App) exportFile(outPath, filePath string, zipWr *zip.Writer) *model.AppError {
+func (a *App) exportFile(outPath, filePath string, zipWr *zip.Writer, rctx request.CTX ) *model.AppError {
```
Comment (hanzei) [inline server/channels/app/export.go:1145]: ` request.CTX` should always be the first parameter. Can you please update the code?

File: server/channels/app/export.go [hunk 6]
```diff
@@ -897,12 +902,16 @@ func (a *App) exportCustomEmoji(c request.CTX, job *model.Job, writer io.Writer,
...
 	fromPath, err := os.Open(emojiImagePath)
 	if fromPath == nil || err != nil {
 		return errors.New("Error reading " + emojiImagePath + "file")
 	}
-	defer fromPath.Close()
+	defer func() {
+        if err = fromPath.Close(); err != nil {
+            rctx.Logger().Error("Error closing source file", mlog.String("path", emojiImagePath), mlog.Err(err))
+        }
+    }()
```
Comment (hanzei) [inline server/channels/app/export.go:914]: The formatting looks off

**Review by hanzei — CHANGES_REQUESTED** [review]

**Review by hanzei — CHANGES_REQUESTED** [review]
The formatting still looks off. Can you please run `go fmt channels/app/export.go` to fix that?

**Comment by TheInvincibleRalph** [comment]
> The formatting still looks off. Can you please run `go fmt channels/app/export.go` to fix that?
image

The formatter is flagging a `missing ',' in argument lists` in the function in the image, and that's perhaps why the test is failing. I really don't see anything wrong with that function or the argument list. What do you think?

File: server/channels/app/export.go [hunk 7]
```diff
@@ -920,8 +929,11 @@ func (a *App) copyEmojiImages(emojiId string, emojiImagePath string, pathToDir s
 	if err != nil {
 		return errors.New("Error creating the image file " + err.Error())
 	}
-	defer toPath.Close()
-
+	defer func() {
+		if err = toPath.Close(); err != nil {
+			rctx.Logger().Error("Error closing destination file", mlog.String("path", emojiDir+"/image"), mlog.Err(err))
+        }
+    }()
```
Comment (hanzei) [inline server/channels/app/export.go:936]: Are two curvy braces really correct here?

**Comment by hanzei** [comment]
@TheInvincibleRalph The code doesn't compile. There is a syntax error.

**Comment by TheInvincibleRalph** [comment]
> @TheInvincibleRalph The code doesn't compile. There is a syntax error.

I have made some changes. I hope it compiles this time.

**Review by hanzei — CHANGES_REQUESTED** [review]
There is a compiler error. Can you please take a look? Thanks!

**Comment by TheInvincibleRalph** [comment]
> There is a compiler error. Can you please take a look? Thanks!

Fixed.

File: server/channels/app/export_test.go [hunk 8]
```diff
@@ -156,7 +157,7 @@ func TestCopyEmojiImages(t *testing.T) {
 	os.OpenFile(filePath+"/image", os.O_RDONLY|os.O_CREATE, 0777)
 	defer os.RemoveAll(filePath)
 
-	copyError := th.App.copyEmojiImages(emoji.Id, emojiImagePath, pathToDir)
+	copyError := th.App.copyEmojiImages(request.CTX, emoji.Id, emojiImagePath, pathToDir)
```
Comment (hanzei) [inline server/channels/app/export_test.go:160]: That is not valid code. You can use `th.Context` instead.

File: server/i18n/en.json [hunk 9]
```diff
@@ -10408,4 +10412,4 @@
     "id": "web.incoming_webhook.user.app_error",
     "translation": "Couldn't find the user."
   }
-]
+]
```
Comment (hanzei) [inline server/i18n/en.json:10415]: Please remove the unrelated change.

**Review by hanzei — CHANGES_REQUESTED** [review]

**Comment by hanzei** [comment]
@nickmisasi, I am removing you from the reviewers list for now while I sort out the CI issues.

File: server/channels/app/export_test.go (hunk 8 above)
Comment (TheInvincibleRalph) [reply server/channels/app/export_test.go:160]: > That is not valid code. You can use `th.Context` instead.

Ooops.

File: server/channels/app/export.go [hunk 10]
```diff
@@ -1155,15 +1171,19 @@ func (a *App) exportFile(outPath, filePath string, zipWr *zip.Writer) *model.App
 				nil, "err="+err.Error(), http.StatusInternalServerError)
 		}
 
-		wr, err = os.Create(filePath)
+		wr, err := os.Create(filePath)
```
Comment (hanzei) [inline server/channels/app/export.go:1174]: No need for a new variable
```suggestion
		wr, err = os.Create(filePath)
```

**Review by hanzei — CHANGES_REQUESTED** [review]

**Review by hanzei — APPROVED** [review]
LGTM :rocket:

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks @TheInvincibleRalph
